prov/efa/src/rxr/rxr_msg.c

```diff
@@ -157,12 +157,12 @@ ssize_t rxr_msg_post_rtm(struct rxr_ep *rxr_ep, struct rxr_tx_entry *tx_entry)
 		 * the information whether the peer
 		 * support it or not.
 		 */
-		err = rxr_pkt_wait_handshake(rxr_ep, tx_entry->addr, peer);
+		err = rxr_pkt_trigger_handshake(rxr_ep, tx_entry->addr, peer);
 		if (OFI_UNLIKELY(err))
 			return err;
 
-		assert(peer->flags & RXR_PEER_HANDSHAKE_RECEIVED);
-		if (!rxr_peer_support_delivery_complete(peer))
+		if (peer->flags & RXR_PEER_HANDSHAKE_RECEIVED &&
```

Comment:
I think we need a local variable ```queue_tx``` and set it like

```
queue_tx = false;
if (!(peer->flags & RXR_PEER_HANDSHAKE_RECEIVED)) {
    queue_tx = true;
} else if(rxr_peer_support_delivery_complete(peer)) {
    queue_tx = false;
}  else{
    return -FI_EOTHER;
}
```
So it is more clear that before we receive handshake, we will queue tx_entry.

Reply:
queueing logic is no longer used, resolving.